actor/v2action/composite/service_broker_summary.go

```diff
@@ -36,13 +40,44 @@ type ServiceBrokerSummaryCompositeActor struct {
 	VisibilityActor VisibilityActor
 }
 
-func (c *ServiceBrokerSummaryCompositeActor) GetServiceBrokerSummaries(brokerName string, service string, organization string) ([]v2action.ServiceBrokerSummary, v2action.Warnings, error) {
+// GetServiceBrokerSummaries returns summaries for service brokers that match the arguments passed.
+// An error will be returned if any of the options are invalid (i.e. there is no broker, service or
+// organization with the given names). Each of the provided arguments will act as a filtering mechanism,
+// with the expectation that the caller does not want matches for "parent" concepts, if they have no
+// children matching a filtered argument.
+//
+// For example, given a Broker "Foo", containing a Service "Bar":
+//
+// `GetServiceBrokerSummaries("Foo", "NotBar", "")` will return a slice of broker summaries that does not
+// include the Broker "Foo".
+//
+// Similarly, given a broker "Foo", containing a Service "Bar", that has plans available only in Organization "Baz":
+//
+// `GetServiceBrokerSummaries("Foo", "Bar", "NotBaz") will recurse upwards resulting in a slice of broker
+// summaries that does not include the Broker "Foo" either.
+func (c *ServiceBrokerSummaryCompositeActor) GetServiceBrokerSummaries(brokerName string, serviceName string, organizationName string) ([]v2action.ServiceBrokerSummary, v2action.Warnings, error) {
 	var (
 		err      error
 		warnings v2action.Warnings
 		brokers  []v2action.ServiceBroker
 	)
 
+	if organizationName != "" {
+		orgExistsWarnings, orgExistsErr := c.ensureOrgExists(organizationName)
+		warnings = append(warnings, orgExistsWarnings...)
+		if orgExistsErr != nil {
+			return nil, warnings, orgExistsErr
+		}
+	}
+
+	if serviceName != "" {
+		serviceExistsWarnings, serviceExistsErr := c.ensureServiceExists(serviceName)
```

Comment:
It wasn't. but on more thought about the scenario, it would be when both broker and service filters are provided, broker exists and the service exists but not for the broker provided. With the current implementation, it would print an empty list. Taking this one step further, similar scenarios also exist for org+broker, org+service and org+broker+service filter combinations. And handling all of them is going to complicate this already complex implementation. 
@abbyachau @laurzca Wondering if an empty list is acceptable in these scenarios or do we want to explicitly fail?